src/formats/formats.js

```diff
@@ -37,40 +37,42 @@ export function formatByExtension(extension) {
   }[extension];
 }
 
-function formatByName(name) {
+function formatByName(name, customDelimiter) {
   return {
     yml: yamlFormatter,
     yaml: yamlFormatter,
     toml: tomlFormatter,
     json: jsonFormatter,
     frontmatter: FrontmatterInfer,
-    'json-frontmatter': FrontmatterJSON,
-    'toml-frontmatter': FrontmatterTOML,
-    'yaml-frontmatter': FrontmatterYAML,
+    'json-frontmatter': frontmatterJSON(customDelimiter),
+    'toml-frontmatter': frontmatterTOML(customDelimiter),
+    'yaml-frontmatter': frontmatterYAML(customDelimiter),
   }[name];
 }
 
 export function resolveFormat(collectionOrEntity, entry) {
+  // Check for custom delimiter
+  const customDelimiter = collectionOrEntity.get('frontmatter_delimiter');
   // If the format is specified in the collection, use that format.
-  const format = collectionOrEntity.get('format');
-  if (format) {
-    return formatByName(format);
+  const formatSpecification = collectionOrEntity.get('format');
+  if (formatSpecification) {
+    return formatByName(formatSpecification, customDelimiter);
   }
 
   // If a file already exists, infer the format from its file extension.
   const filePath = entry && entry.path;
   if (filePath) {
     const fileExtension = filePath.split('.').pop();
-    return formatByExtension(fileExtension);
+    return formatByExtension(fileExtension, customDelimiter);
```

Comment:
I don't think we need to pass `customDelimiter` to `formatByExtension()` when it doesn't use it. That can always be added later if needed.

Reply:
I just removed the unnecessary passing of customDelimiter to formatByExtension(), it was a remnant of a different train of thought. Sorry I didn't have it cleaned up before.  

Are there any other steps I should take with this?

Reply:
Nope, thank you.